PictureApp/Controller/PicturesViewController.swift

```diff
@@ -0,0 +1,306 @@
+        
+        
+    }
+    
+     
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        let buttonOptions = UIButton(type: .contactAdd)
+        buttonOptions.addTarget(self, action: #selector(showOptions(_:)), for: .touchUpInside)
+        navigationItem.rightBarButtonItem = .init(customView: buttonOptions)
+        setupCollectionView()
+
+        fetchHits(order: orderForCategories,selectedCategory: ["music"],currentPage: currentPage)
+        
+    }
+
+}
+    
+extension PicturesViewController : UICollectionViewDelegate,UICollectionViewDataSource {
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return hits.count
+    }
+  
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: GalleryCollectionViewCell.identifier, for: indexPath) as! GalleryCollectionViewCell
+        
+        let tagText = hits[indexPath.item].tags
+      
+        let url = URL(string: hits[indexPath.row].image)!
+        ImageLoader(with: URLHTTPClient(session: URLSession.shared)).load(from: url) { (image) in
+            DispatchQueue.main.async {
+                guard collectionView.indexPath(for: cell) == indexPath else {
+
+                    return }
+
+                cell.imageView.image = image
+                cell.tagsLabel.text = tagText
+        
+            }
+        }
+        
+        return cell
+    }
+    
+    
+    //MARK:Pagination
+    func collectionView(_ collectionView: UICollectionView, willDisplay cell: UICollectionViewCell, forItemAt indexPath: IndexPath) {
+        if hits.count - 1 < totalHits && indexPath.row == hits.count - 1 {
+            currentPage = currentPage + 1
+            
+            fetchHits(order: orderForCategories, selectedCategory: selectedCategoryForSort,currentPage: currentPage)
+           
+        }
+    }
+    
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+      
+        let controller = storyboard!.instantiateViewController(withIdentifier: "PreviewViewController") as! PreviewViewController
+        controller.title = "Preview"
+        let selectedPicture = hits[indexPath.item]
+        let tagsText = hits[indexPath.item].tags
+       
+       
+        controller.configureUI(with: selectedPicture, tagText: tagsText)
+        navigationController?.pushViewController(controller, animated: true)
+        
+    }
+    
+}
+
+
+extension PicturesViewController : UICollectionViewDelegateFlowLayout{
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        if UIDevice.current.orientation.isLandscape {
+        landscapeCollectionViewLayout()
+            
+        }else if UIDevice.current.orientation.isPortrait {
```

Comment:
Do we need to use this instead of just `else`?

Reply:
"Else" would be enough. Done